modules/monitoring-deploying-a-sample-application.adoc

```diff
@@ -0,0 +1,99 @@
+// Module included in the following assemblies:
+//
+// * monitoring/user-workload-monitoring.adoc
+
+[id="deploying-a-sample-application_{context}"]
+= Deploying a sample application
+
+To test user workload monitoring, you can deploy a sample application.
+
+.Procedure
+
+. Create a YAML file for the application configuration. In this example, it is called `prometheus-example-app.yaml`.
+
+. Fill the file with the configuration for deploying the application:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: ns1
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    app: prometheus-example-app
+  name: prometheus-example-app
+  namespace: ns1
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: prometheus-example-app
+  template:
+    metadata:
+      labels:
+        app: prometheus-example-app
+    spec:
+      containers:
+      - image: quay.io/brancz/prometheus-example-app:v0.1.0
+        imagePullPolicy: IfNotPresent
+        name: prometheus-example-app
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: prometheus-example-app
+  name: prometheus-example-app
+  namespace: ns1
+spec:
+  ports:
+  - port: 8080
+    protocol: TCP
+    targetPort: 8080
+    name: web
+  selector:
+    app: prometheus-example-app
+  type: ClusterIP
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    k8s-app: prometheus-example-monitor
+  name: prometheus-example-monitor
+  namespace: ns1
+spec:
+  endpoints:
+  - interval: 30s
+    port: web
+    scheme: http
+  selector:
+    matchLabels:
+      app: prometheus-example-app
+----
++
+This configuration deploys an application named `prometheus-example-app` in the `ns1` namespace. This application exposes the custom `version` metric.
+
+. Apply the configuration file to the cluster:
++
+----
+$ oc apply -f prometheus-example-app.yaml
+----
++
+It will take some time to deploy the application.
+
+. You can check that the application is running and exposing metrics:
++
+----
+$ oc -n ns1 get pod
+NAME                                      READY     STATUS    RESTARTS   AGE
+prometheus-example-app-7857545cb7-sbgwq   1/1       Running   0          81m
+
+$ oc -n ns1 get servicemonitor
+NAME                         AGE
+prometheus-example-monitor   81m
+----
```

Comment:
We should also give an example of a recording rule/alert including the current limitations:

```
apiVersion: monitoring.coreos.com/v1
kind: PrometheusRule
metadata:
  name: example-alert
  namespace: ns1
spec:
  groups:
  - name: example
    rules:
    - alert: VersionAlert
      expr: version{job="prometheus-example-app"} == 0
```

Note: The expression can only reference metrics exposed via user workload monitoring. Currently it is not possible to correlate existing cluster level metrics.

Reply:
Added as a separate "Creating an alerting rule for user workload metrics" module.